programs/compressed-token/src/token_data.rs

```diff
@@ -0,0 +1,327 @@
+use std::vec;
+
+use anchor_lang::{
+    prelude::borsh, solana_program::pubkey::Pubkey, AnchorDeserialize, AnchorSerialize,
+};
+use light_hasher::{errors::HasherError, DataHasher};
+use light_utils::hash_to_bn254_field_size_be;
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, AnchorSerialize, AnchorDeserialize)]
+#[repr(u8)]
+pub enum AccountState {
+    Initialized,
+    Frozen,
+}
+
+#[derive(Debug, PartialEq, Eq, AnchorSerialize, AnchorDeserialize, Clone, Copy)]
+pub struct TokenData {
+    /// The mint associated with this account
+    pub mint: Pubkey,
```

Comment:
`TokenData` and `TokenData` are almost identical, I can spot only one different field. Any chance we could move the common fields to a new struct and include it as fields?

Reply:
yeah this is on purpose. Anchor ts enums didn't work out of the box, hence the workaround to have a TokenDataClient struct that is explicit about the enum being a u8.
We should figure out how to handle the enum on ts side and remove the TokenDataClient then.